standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -1917,6 +1923,12 @@ private void drop_database_core(RawStore ms, String catName,
             LOG.error("Failed to delete database directory: " + db.getLocationUri() +
                 " " + e.getMessage());
           }
+          try {
+            wh.deleteDir(wh.getDatabaseManagedPath(db), true, db);
+          } catch (Exception e) {
+            LOG.error("Failed to delete database directory: " + db.getLocationUri() +
```

Comment:
should we are more consistent with the delete behavior, throw exception here too (the same as line 1789), the same is for 1923

Reply:
I will rebase this patch. All this code has been refactored into HMSHandler.java. I will have to redo this fix. I will address it then.